### PR TITLE
Implemented private route component

### PR DESCRIPTION
**Pull request by mctran-phi**
### What this PR does (required):
- Implemented a private route component to only allow authenticated users access.

### Screenshots / Videos (required):
- N/A

### Any information needed to test this feature (required):
- Access a private route without being logged in, it should redirect you to the login page.

### CODE REVIEW

File: client/src/components/PrivateRoute/PrivateRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+const PrivateRoute: React.FC<{
+  component: React.FC;
+  path: string;
+  exact: boolean;
+  loggedIn: boolean;
+}> = (props) => {
+  const loggedIn = props.loggedIn;
```
Comment (rajivtitus) [inline client/src/components/PrivateRoute/PrivateRoute.tsx]: Maybe get the logged in user using the auth context (useAuth hook)

File: client/src/components/PrivateRoute/PrivateRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+const PrivateRoute: React.FC<{
+  component: React.FC;
```
Comment (rajivtitus) [inline client/src/components/PrivateRoute/PrivateRoute.tsx:6]: Please create a separate interface and assign it to props :)

**Review by rajivtitus — COMMENTED** [review]
Just a couple of small comments, please have a look :)

**Comment by mctran-phi** [comment]
Made the changes, please take a look.

File: client/src/components/PrivateRoute/PrivateRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { useAuth } from '../../context/useAuthContext';
+
+interface Props {
+  component: React.FC;
+  path: string;
+  exact: boolean;
+}
+
+const PrivateRoute:React.FC<Props> = (props:Props) => {
+  const authUser = useAuth();
```
Comment (rajivtitus) [inline client/src/components/PrivateRoute/PrivateRoute.tsx]: Additionally, you could also destructure the object here `{loggedInUser}`  :)

**Review by rajivtitus — APPROVED** [review]
Thanks for making changes. Looks good to me! :)

**Pull request merged**